Allow users to pass arguments to the stf perl layer

adamfarley opened the pull request:
If the user wants to use stf.pl options that we don't anticipate in
the makefile, this option allows them to do that. This prevents code
duplication and dependencies between the make and perl code.

Signed-off-by: Adam Farley <adam.farley@uk.ibm.com>

Commit:
Allow users to pass arguments to the stf perl layer

If the user wants to use stf.pl options that we don't anticipate in
the makefile, this option allows them to do that. This prevents code
duplication and dependencies between the make and perl code.

Signed-off-by: Adam Farley <adam.farley@uk.ibm.com>

lumpfish commented:
"This prevents code duplication and dependencies between the make and perl code." It also opens up the possibility that people will try to run tests with options which interfere with the successful execution of the test.
What was the actual issue which led you to propose this change?

adamfarley commented:
The core issue was a new feature added to stf, which automatically deletes the results files from a passed run. I considered adding a make argument specifically for this stf.pl argument, but realised that this would require a new make variable *every time*. Plus, we'd need to remember to modify the makefile whenever we modified the stf.pl option.

lumpfish commented:
So an stf argument such as -delete-if-passed and then the means to pass that through from a makefile invocation such as DELETE_IF_PASSED=1.  An issue with just using STF_ARGS=DELETE_IF_PASSED=1 is that there is nothing to prevent a user from combining multiple options.  This will lead to issues in the makefile of how to deal with whitespace and quotes such that the arguments are in the form required on the stf command line.

adamfarley commented:
Then I'll modify this PR, and the openjdk-systemtest PR, to be option-specific. This means they shouldn't be merged until https://github.com/AdoptOpenJDK/stf/pull/14 is approved and merged, so I suggest reviewing that while I put the option-specific changes into this PR.

Commit:
Enable automatic deletion of results files if the test passes

If the test passes, using this option with automatically delete 
the results directory, leaving more hard disk space for results
files from failed tests.

Signed-off-by: Adam Farley <adam.farley@uk.ibm.com>

lumpfish commented:
Ref https://github.com/AdoptOpenJDK/stf/issues/10
